pkg/runtime/reconciler.go

```diff
@@ -580,8 +581,46 @@ func (r *resourceReconciler) lateInitializeResource(
 	return lateInitializedLatest, err
 }
 
-// patchResourceMetadataAndSpec patches the custom resource in the Kubernetes API to match the
-// supplied latest resource's metadata and spec.
+// getPatchDocument returns a JSON string containing the object that will be
+// patched in the Kubernetes API server.
+//
+// NOTE(jaypipes): Because the Kubernetes API server's server-side apply
+// functionality introduces an enormous amount of verbose annotations in the
+// resource metadata, and because those annotations are pretty unhelpful to
+// ACK, we strip all that stuff out of the returned patch document.
+func getPatchDocument(
+	patch client.Patch,
+	obj client.Object, // the diff of this will be represented in the patch
+) string {
+	js, _ := patch.Data(obj)
+	var m map[string]interface{}
+	_ = json.Unmarshal(js, &m)
+	if md, ok := m["metadata"]; ok {
+		// Strip out managedFields stuff, since it's super verbose and
+		// doesn't offer any value to us (since we don't use server-side
+		// apply
+		if mv, ok := md.(map[string]interface{}); ok {
```

Comment:
Is it necessary to check the `md` type here? i suppose it's always going to be a `map[string]interface{}`

Reply:
actually `md` is `interface{}` not `map[string]interface{}` since `md` is an element of the top-level `map[string]interface{}` :)